### PR TITLE
[Bugfix / Feature] force user to wait until upload is completed and allow user to select delete multiple files

### PR DESCRIPTION
**Pull request by karel-harjono**
Fixes issue https://github.com/ubco-db/helpme/issues/20, please visit issue for more details

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Any UI changes have been checked to work on desktop, tablet, and mobile

image
image
image
image

### CODE REVIEW

File: packages/app/components/Settings/ChatbotSettings.tsx [hunk 1]
```diff
@@ -172,9 +239,10 @@ export default function ChatbotSettings(): ReactElement {
           headers: { HMS_API_TOKEN: profile.chat_token.token },
         })
 
-        toast.success('File uploaded.')
+        toast.success(`${file.name} uploaded.`)
```
Comment (wskksw) [inline packages/app/components/Settings/ChatbotSettings.tsx:242]: I realized that there is an error here. This call doesn't check status response and does toast.success even though a 500 server error is returned. 
It's not caused by your changes, but I wonder can you edit it?

**Review by wskksw — COMMENTED** [review]
I have requested for a small change for historical errors. 
Overall your new changes make sense and I like the UI. 
Interestingly, there can be a different way of potentially dealing with asynchronous tasks. I have recently come across OpenAI's strategies, see the polling and batching helpers of this link. [https://github.com/openai/openai-node?tab=readme-ov-file#polling-helpers](url)

File: packages/app/components/Settings/ChatbotSettings.tsx (hunk 1 above)
Comment (karel-harjono) [reply packages/app/components/Settings/ChatbotSettings.tsx:242]: I see what you mean, I will update. so here's what I'll do, if upload for a given document fail, toast.failure("{doc1} upload failed") and go on with the rest of the process(delete doc2, doc3,...). let me know if you have any other suggestion

**Review by wskksw — APPROVED** [review]

File: packages/app/components/Settings/ChatbotSettings.tsx (hunk 1 above)
Comment (wskksw) [reply packages/app/components/Settings/ChatbotSettings.tsx:242]: Sounds good, and that's it

**Pull request merged**